tests/tools/onert_train/src/rawdataloader.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "rawdataloader.h"
+#include "nnfw_util.h"
+
+#include <iostream>
+#include <stdexcept>
+#include <numeric>
+
+namespace onert_train
+{
+
+Generator RawDataLoader::loadDatas(const std::string &input_file, const std::string &expected_file,
+                                   std::vector<nnfw_tensorinfo> &input_infos,
+                                   std::vector<nnfw_tensorinfo> &expected_infos,
+                                   const uint32_t data_length, const uint32_t batch_size)
+{
+  auto size_accumulator = [](const unsigned int &a, const nnfw_tensorinfo &b) {
+    return a + bufsize_for(&b);
+  };
+
+  std::vector<uint32_t> input_origins(input_infos.size());
+  uint32_t start = 0;
+  for (uint32_t i = 0; i < input_infos.size(); ++i)
+  {
+    input_origins.at(i) = start;
+    start += (bufsize_for(&input_infos[i]) * data_length);
+  }
+
+  std::vector<uint32_t> expected_origins(expected_infos.size());
+  start = 0;
+  for (uint32_t i = 0; i < expected_infos.size(); ++i)
+  {
+    expected_origins.at(i) = start;
+    start += (bufsize_for(&expected_infos[i]) * data_length);
+  }
```

Comment:
This code assumes the data will be like below:

num_input : 3
data_length : 5

```
# in(input num)_(data num)

in0_0 in0_1 in0_2 in0_3 in0_4 in1_0 in1_1 in1_2 in1_3 in1_4 in2_0 in2_1 in2_2 in2_3 in2_4
```